### PR TITLE
add keepalived

### PR DESCRIPTION
**Pull request by lfchris**

### CODE REVIEW

File: roles/keepalived/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+keepalived__password: 'password'
+keepalived__virtual_ipaddress: '192.168.56.199'
+```
+
+
+## Optional Role Variables
+
+| Variable | Description | Default Value |
+| -------- | ----------- | ------------- |
+| `keepalived__interface` | The network interface keepalived should use. | `default ipv4 interface` |
```
Comment (NavidSassan) [inline roles/keepalived/README.md:41]:
```suggestion
| `keepalived__interface` | The network interface keepalived should use. | `{{ ansible_facts["default_ipv4"]["interface"] }}` |
```
only use an explanation if its not clear, or the default is too long. and if using an explanation, do not enclose it in backticks.

File: roles/keepalived/handlers/main.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+- name: 'keepalived: restart keepalived'
+  systemd:
```
Comment (NavidSassan) [inline roles/keepalived/handlers/main.yml:2]: use FQCN

File: roles/keepalived/templates/etc/keepalived/keepalived.conf.j2 [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{{ ansible_managed }}
```
Comment (NavidSassan) [inline roles/keepalived/templates/etc/keepalived/keepalived.conf.j2:1]: no comment needed? what is the comment char? `#` or `!`?

File: roles/keepalived/templates/etc/keepalived/keepalived.conf.j2 [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+     {% endfor %}
+   }
+   notification_email_from {{ keepalived__notification_email_from }}
+   smtp_server {{ keepalived__smtp_server }}
+   smtp_connect_timeout 30
+}
+
+vrrp_instance VI_1 {
+        state {{ keepalived__state }}
+        interface {% if keepalived__interface is defined and keepalived__interface | length %}{{ keepalived__interface }}{% else %}{{ ansible_facts["default_ipv4"]["interface"] }}{% endif %}
```
Comment (NavidSassan) [inline roles/keepalived/templates/etc/keepalived/keepalived.conf.j2:19]: why not just set `keepalived__interface: {{ ansible_facts["default_ipv4"]["interface"] }}` in the defaults?

File: roles/keepalived/README.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+| `keepalived__virtual_ipaddress` | The IP address to be shared between the MASTER and BACKUP. |
+
+Example:
+```yaml
+# mandatory
+keepalived__notification_email_addresses:
+  - 'root@example.com'
+keepalived__state: 'MASTER'
+keepalived__password: 'password'
+keepalived__virtual_ipaddress: '192.168.56.199'
```
[the remark this answers is not in the document]
Comment (lfchris) [reply roles/keepalived/README.md:33]: Those examples are nets, no ip addresses.

**Comment by NavidSassan** [comment]
looks good :)

**Pull request merged**